listenbrainz/webserver/static/js/tests/fresh-releases/FreshReleases.test.tsx

```diff
@@ -0,0 +1,62 @@
+import * as React from "react";
+
+import { mount } from "enzyme";
+import GlobalAppContext, { GlobalAppContextT } from "../../src/utils/GlobalAppContext";
+import APIService from "../../src/utils/APIService";
+
+import FreshReleases from "../../src/fresh-releases/FreshReleases";
+
+const freshReleasesProps = {
+  "user": {
+    "name": "chinmaykunkikar",
+    "id": 1
+  },
+  "profileUrl": "/user/chinmaykunkikar/",
+  "spotify": {
+    "access_token": "access-token",
+    "permission": ["streaming", "user-read-email", "user-read-private"]
+  },
+  "youtube": {
+    "api_key": "fake-api-key"
+  }
+}
+
+const { youtube, spotify, user } = freshReleasesProps;
+
+const props = {
+  ...freshReleasesProps,
+  newAlert: () => { },
+};
+
+// Create a new instance of GlobalAppContext
+const mountOptions: { context: GlobalAppContextT } = {
+  context: {
+    APIService: new APIService("foo"),
```

Comment:
I'm not clear on the differences for testing funcional components vs classes.
Here's something you could try if you wanted to mock some API methods (not sure it'll work but…):
Along the lines of
```

const fakeAPIService = new APIService("foo");
const mockFetchSitewideFreshReleases = jest.fn()
fakeAPIService.fetchSitewideFreshReleases = mockFetchSitewideFreshReleases
const wrapper = mount(
  <GlobalAppContext.Provider value={{...mountOptions.context, APIService: fakeAPIService}}>
    <FreshReleases {...props} />
  </GlobalAppContext.Provider>
);
expect(mockFetchSitewideFreshReleases).toHaveBeenCalled()
```

Reply:
Will try it out

Reply:
Update: It works. I'll try to take it forward from here :)

Reply:
Phew! Great!